### ISSUE TITLE
Bug - Z2M 2.0 will break all blueprints in January 2025

### ISSUE THREAD

**Issue by dagstuan** [EPMatt/awesome-ha-blueprints#622]
### Blueprint name

All blueprints

### Home Assistant Core Version

-

### Home Assistant Installation Type

Home Assistant Core

### Description

As mentioned in the latest [release notes](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.42.0): Zigbee2MQTT 2.0 is planned to be released in January 2025. It will remove the `homeassistant_legacy_triggers` currently required to trigger these blueprints. Hopefully they can be updated in time.

Zigbee2MQTT docs suggest moving to MQTT device triggers instead of the `sensor.*`-triggers. Docs for breaking changes are [here](https://github.com/Koenkk/zigbee2mqtt/discussions/24198).

I saw #244 but I figured it would be nice with a new issue for visibility, since that one is quite old.

### Automation YAML config

```yaml
-
```

### To Reproduce

Upgrade to Z2M 2.0

### Expected behavior

Blueprints supporting MQTT device triggers.

### Actual Behaviour

Currently no support for MQTT device triggers.

### Additional Details

- [ ] I'd like to help developing a fix for this issue.

**Comment by ant1fr** [comment]
Hi. Any news regarding this bug? I dropped the legacy mode in Z2M in preparation for Z2M 2.0...

**Comment by oskari** [comment]
Probably wise to start migrating away, as the repo seems to be dead for quite a while now: https://github.com/EPMatt/awesome-ha-blueprints/issues/554

**Comment by ant1fr** [comment]
Thanks @oskari . But sad to hear :(

**Comment by yarafie** [comment]
There is a fork that is working on updating the blueprints (controllers) to work with z2m 2.0.0

https://github.com/lsismeiro/awesome-ha-blueprints/issues/26

BTW: Currently z2m devs are delaying the removal of legacy triggers from the 2.0.0 version

**Comment by EPMatt** [comment]
Hey there,
first of all thanks everyone for your kind patience - life happens, and unfortunately I couldn't contribute at all to this project for a long time. Luckily, I'm hopefully now slowly getting back to be more active here.

Thank you all for your continuous support - can't believe we are getting close to 900 stars! Let's bring Awesome HA Blueprints back to life! 🔥

**Comment by EPMatt** [comment]
I might get to work on this in the following days. It looks like someone opened a PR some time ago with support for Direct Triggers for one of our blueprints.

https://github.com/EPMatt/awesome-ha-blueprints/pull/602

Will replicate the change to all blueprints, making sure we provide backward compatibility to people still on Z2M <2.0.0.

Anyone here willing to help me test out the fix for at least a few controllers once the PR is ready?

**Comment by yarafie** [comment]
@EPMatt 

There is a fork that is working on updating the blueprints (controllers) to work with z2m 2.0.0

[https://github.com/lsismeiro/awesome-ha-blueprints/issues/26](url)

BTW: Currently z2m devs are delaying the removal of legacy triggers from the 2.0.0 _version_

**Comment by yarafie** [comment]
> I might get to work on this in the following days. It looks like someone opened a PR some time ago with support for Direct Triggers for one of our blueprints.
> 
> #602
> 
> I will replicate the change to all blueprints, also making sure we provide backward compatibility to people still on Z2M <2.0.0.
> 
> Anyone here willing to help me test out the fix for at least a few controllers once the PR is ready?

@EPMatt 

There is a fork that is working on updating the blueprints (controllers) to work with z2m 2.0.0 with backwards compatibility.

https://github.com/lsismeiro/awesome-ha-blueprints

BTW: Currently z2m devs are delaying the removal of legacy triggers from the 2.0.0 _version_

**Comment by hvorragend** [comment]
I can warmly recommend this fork.
I myself also use the repositories from yarafie and lsismeiro in my HA installations.

**Comment by EPMatt** [comment]
Thanks for pointing me out to the fork. Love the work you've done there guys!

I totally understand how upsetting it must have been for you all, users and supporters, to see this project fall into oblivion. I'm honestly left speechless by looking at the efforts and time everyone here put to keep the project alive while I was away. This simply shows how much you care about Awesome HA Blueprints, and motivates me to bring my contributions and time back to this project. Thank you so much! 🔥 

I can see the fix for this bug is going strong in the fork. So, I'd be really happy to join efforts and gradually bring the development back to the main repo here. Let's discuss how to better approach this!

**Comment by richie80** [comment]
You can 'fix' your blueprints after migrating to 2.0.0 with checking this box: 
![image](https://github.com/user-attachments/assets/c5cb36ca-aaaf-4b46-b97d-68ad51ff200f)

**Comment by SHxKM** [comment]
Thanks. But doesn't the light hook depend on entity attributes which are also gone?

**Comment by d33c0n** [comment]
@SHxKM the setting @richie80 mentioned in Zigbee2Mqtt brings them back (for now).

**Comment by yarafie** [comment]
A few of us are working on a solutiom to prepare for z2m 2.x that will also be backwards compatible with z2m 1.x.

**Comment by SHxKM** [comment]
> @SHxKM the setting @richie80 mentioned in Zigbee2Mqtt brings them back (for now).

Action sensors are not the same as entity attributes.

**Comment by theincogtion** [comment]
Just updated Z2m and this one is broken:
https://epmatt.github.io/awesome-ha-blueprints/docs/blueprints/controllers/ikea_e2001_e2002/

**Comment by d33c0n** [comment]
@SHxKM sorry, missed that you were talking about attributes, but I guess then the answer to your question is no. The light and control hook need the action sensor, which is being brought back by that setting, have it working on my end.

**Comment by SHxKM** [comment]
> @SHxKM sorry, missed that you were talking about attributes, but I guess then the answer to your question is no. The light and control hook need the action sensor, which is being brought back by that setting, have it working on my end.

That's interesting, because this is what I get when I disable the legacy attributes in Z2M:

```
Template variable warning: None has no element 0 when rendering '{{ [((state_attr(light,"hs_color")[0] or 0) + 15) % 360, 100] }}'
```

And as you can see, this is part of the light hook:

https://github.com/EPMatt/awesome-ha-blueprints/blob/20ef4e40e808447b07b9546ed473d8fdfcbd34aa/blueprints/hooks/light/light.yaml#L528-L532

**Comment by catch-33** [comment]
> You can 'fix' your blueprints after migrating to 2.0.0 with checking this box: ![image](https://private-user-images.githubusercontent.com/61156727/400122886-c5cb36ca-aaaf-4b46-b97d-68ad51ff200f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwMDY4MDMsIm5iZiI6MTczNjAwNjUwMywicGF0aCI6Ii82MTE1NjcyNy80MDAxMjI4ODYtYzVjYjM2Y2EtYWFhZi00YjQ2LWI5N2QtNjhhZDUxZmYyMDBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA0VDE2MDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NmMxNTIyZTgxNzRlYWJmNDY5MDQ1OTYxNDk5MTA4MGI0YTVjODBmOTBlNTkzMGIzNjNkZWU5ZjliOTJhNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AFIaBogTARSrnDHoEmQYUyucyFEpIK4A_8akm6XarEA)

Wherre can I find this setting?

**Comment by SHxKM** [comment]
> > You can 'fix' your blueprints after migrating to 2.0.0 with checking this box: ![image](https://private-user-images.githubusercontent.com/61156727/400122886-c5cb36ca-aaaf-4b46-b97d-68ad51ff200f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwMDY4MDMsIm5iZiI6MTczNjAwNjUwMywicGF0aCI6Ii82MTE1NjcyNy80MDAxMjI4ODYtYzVjYjM2Y2EtYWFhZi00YjQ2LWI5N2QtNjhhZDUxZmYyMDBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA0VDE2MDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1NmMxNTIyZTgxNzRlYWJmNDY5MDQ1OTYxNDk5MTA4MGI0YTVjODBmOTBlNTkzMGIzNjNkZWU5ZjliOTJhNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AFIaBogTARSrnDHoEmQYUyucyFEpIK4A_8akm6XarEA)
> 
> Wherre can I find this setting?

`Zigbee2MQTT → [Cog Icon] → Home Assistant Integration`

**Comment by d33c0n** [comment]
@SHxKM I double checked on my end, that attribute is still there in Z2M 2.0 even with the legacy action sensor option in Z2M unchecked:


<img width="820" alt="image" src="https://github.com/user-attachments/assets/2031bbe3-2c94-46db-af9e-e57eb11d2bd7" />

**Comment by SHxKM** [comment]
> @SHxKM I double checked on my end, that attribute is still there in Z2M 2.0 even with the legacy action sensor option in Z2M unchecked:

That's because it goes away after you change `homeassistant_legacy_entity_attributes` to `false`. Which is precisely my point: actions aren't the only thing that will be missing.

**Comment by d33c0n** [comment]
@SHxKM do you happen to have some documentation about these options? I am confused by now. I didn't really dig into this, but in my understanding these options were only needed to prep your Z2M <2.0 for the upgrade, I believe they were even automatically removed from the config.yml during the update.

Nevertheless, I addedd them back in now and still see all the attributes. I also couldn't find the error message you had in my logs.

**Comment by lyricnz** [comment]
My hooks are all broken (from hue switch) after the upgrade, even with legacy turned on. I suspect the event is missing some data which the blueprint depends on? Regardless, I've switched to a simpler solution https://community.home-assistant.io/t/very-easy-custom-philips-hue-dimmer-switch-for-z2m-zigbee2mqtt/521205/6 and may come back if it gets fixed.

**Comment by yarafie** [comment]
If anyone is up to testing, check this branch

https://github.com/lsismeiro/awesome-ha-blueprints

We are working to add z2m 2.x support and discussions with @EPMatt  are ongoing to see how to merge back into the main branch

**Comment by SHxKM** [comment]
> My hooks are all broken (from hue switch) after the upgrade, even with legacy turned on. I suspect the event is missing some data which the blueprint depends on? Regardless, I've switched to a simpler solution https://community.home-assistant.io/t/very-easy-custom-philips-hue-dimmer-switch-for-z2m-zigbee2mqtt/521205/6 and may come back if it gets fixed.

Same here. Really weird. I don't want to change to the new solution in https://github.com/lsismeiro/awesome-ha-blueprints because I think relying on Device IDs/triggers is not a good solution. I'd rather stay with the deprecated actions for now.

**Comment by SHxKM** [comment]
> @SHxKM sorry, missed that you were talking about attributes, but I guess then the answer to your question is no. The light and control hook need the action sensor, which is being brought back by that setting, have it working on my end.

Is it working with the code in this repo? because I just upgraded to `2.0.0` and nothing is working.

**Comment by lyricnz** [comment]
> Is it working with the code in this repo? because I just upgraded to `2.0.0` and nothing is working. (I enabled legacy actions)

Nope, it's broken for the moment - but I see commits actively being made as I write this. Watch this space.

**Comment by chpego** [comment]
> Thank you all for your continuous support - can't believe we are getting close to 900 stars! Let's bring Awesome HA Blueprints back to life! 🔥

boom! 
![image](https://github.com/user-attachments/assets/07c633fc-eefc-43bd-a4cb-fd488dcbf48c)

**Comment by ctournayre** [comment]
> > @SHxKM sorry, missed that you were talking about attributes, but I guess then the answer to your question is no. The light and control hook need the action sensor, which is being brought back by that setting, have it working on my end.
> 
> Is it working with the code in this repo? because I just upgraded to `2.0.0` and nothing is working. (I enabled legacy actions)

If you enabled legacy actions, you have to relink the controler field to the good controler entity in the 2 automations created from the blueprints

**Comment by EPMatt** [comment]
For anyone looking for a solution without legacy triggers enabled - @yarafie is working on that [here](https://github.com/lsismeiro/awesome-ha-blueprints/pull/33), hopefully we're gonna bring that to you in a PR to this repo soon!

**Comment by EPMatt** [comment]
> boom! ![image](https://private-user-images.githubusercontent.com/38792705/400255202-07c633fc-eefc-43bd-a4cb-fd488dcbf48c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyNDI5ODgsIm5iZiI6MTczNjI0MjY4OCwicGF0aCI6Ii8zODc5MjcwNS80MDAyNTUyMDItMDdjNjMzZmMtZWVmYy00M2JkLWE0Y2ItZmQ0ODhkY2JmNDhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA5MzgwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZTA2MGM3MWRhY2M4NDAzZDYyZGQzY2M5MjU4OTk2NjgyZTlmNDYyMDc2YjI5NTVkMTliYTViNzVkZjQwMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DvbLX-Kc_vK6Cq3r7jqF5CpjWwVdjdnDBbLYWKRP2Us)

Amazing @chpego! 🔥 Thank you all for your incredible support!
Can we make it to 1k? 😎

**Comment by arsenicks** [comment]
Just wanted to chip in with a temporary solution, I've put the opposite of what was suggested in [Zigbee2MQTT 2.0.0 breaking changes #24198](https://github.com/Koenkk/zigbee2mqtt/discussions/24198) so false to true and everything work like it was before the upgrade.

From the doc:
```
In order to get the minimal amount of breaking changes when upgrading from 1.x.x to 2.0.0, make sure you have the following in your Zigbee2MQTT configuration.yaml:

advanced:
  homeassistant_legacy_entity_attributes: false
  homeassistant_legacy_triggers: false
  legacy_api: false
  legacy_availability_payload: false
device_options:
  legacy: false
```

change false to true, I don't know if all those options are required but at least it's a fix for now untill you guy's can upgrade the code!

**Comment by GamesBondNL** [comment]
> Just wanted to chip in with a temporary solution, I've put the opposite of what was suggested in [Zigbee2MQTT 2.0.0 breaking changes #24198](https://github.com/Koenkk/zigbee2mqtt/discussions/24198) so false to true and everything work like it was before the upgrade.

Thanks!  Will give that a try over the weekend. The recommendation to switch to 2.0.0 cold turkey and disable all the legacy stuff seems a bit rigorous for a non-developer setup

**Comment by arsenicks** [comment]
> Thanks! Will give that a try over the weekend. The recommendation to switch to 2.0.0 cold turkey and disable all the legacy stuff seems a bit rigorous for a non-developer setup

Yeah it is but this message was back in october so I think the idea was, disable all the legacy stuff so you'll see what will break before 2.0 comes in. As we can see not many of us did it back then so here we are :P

**Comment by EPMatt** [comment]
For anyone following this thread: Initial PR for MQTT Device triggers for the IKEA E1743 has been opened. 🔥
https://github.com/EPMatt/awesome-ha-blueprints/pull/661

As soon as we merge that, we are gonna follow up with PRs for all the other controllers.
Thanks for everyone who is contributing to the discussion here. Fix is coming so stay tuned! We are gonna need some kind help for testing the updated blueprints when the new PRs for the other controllers are ready, I will notify you here. :)

**Comment by GamesBondNL** [comment]
> For anyone following this thread: Initial PR for MQTT Device triggers for the IKEA E1743 has been opened. 🔥 #661
> 
> As soon as we merge that, we are gonna follow up with PRs for all the other controllers. Thanks for everyone who is contributing to the discussion here. Fix is coming so stay tuned! We are gonna need some kind help for testing the updated blueprints when the new PRs for the other controllers are ready, I will notify you here. :)

Thanks! Will be happy to test E2001/E2002 remotes :)

**Comment by theincogtion** [comment]
E2002 is working fine for me! Waiting for e1810 now 👍

**Comment by GamesBondNL** [comment]
> E2002 is working fine for me! Waiting for e1810 now 👍

Did you enable any of the legacy stuff?  If I disable all the legacy stuff in Z2MQTT 1.4 already E2002 stops working as the sensor.XXX bit is no longer functioning

**Comment by theincogtion** [comment]
> > E2002 is working fine for me! Waiting for e1810 now 👍
> 
> Did you enable any of the legacy stuff? If I disable all the legacy stuff in Z2MQTT 1.4 already E2002 stops working as the sensor.XXX bit is no longer functioning

Yes legend I disabled and I manually had to refund the action sensor in the blueprint

**Comment by GamesBondNL** [comment]
> Version 2025.01.07

Where did you find version 2025.01.07?  The latest in this repository is 2025.01.03

**Comment by theincogtion** [comment]
> > Version 2025.01.07
> 
> Where did you find version 2025.01.07? The latest in this repository is 2025.01.03

From here
https://github.com/lsismeiro/awesome-ha-blueprints/tree/main/blueprints/controllers/ikea_e2001_e2002
But this looks like the old version again. @yarafie 
Not sure why but they are working on merging the repos

**Comment by yarafie** [comment]
Sorry guys.
I'm sure there is lots of confusiion due to the different versions out there.
That's why I deleted my fork of lsismeiro and forked with EPMatt. We R just discussing a few things as we R finding some hurdles trying to avoid breaking changes.
Hope u guys can be a little patient.

**Comment by theincogtion** [comment]
> Sorry guys. I'm sure there is lots of confusiion due to the different versions out there. That's why I deleted my fork of lsismeiro and forked with EPMatt. We R just discussing a few things as we R finding some hurdles trying to avoid breaking changes. Hope u guys can be a little patient.

Thanks for your work and the clarification. 
Patients is hard but I try 😉

**Comment by GamesBondNL** [comment]
> Sorry guys. I'm sure there is lots of confusiion due to the different versions out there. That's why I deleted my fork of lsismeiro and forked with EPMatt. We R just discussing a few things as we R finding some hurdles trying to avoid breaking changes. Hope u guys can be a little patient.

There are a bit many forks, yes!  Thanks for your hard work and efforts

**Comment by EPMatt** [comment]
Update: IKEA E1743 Controller Blueprint Version 2025.01.23 is published, with Z2M 2.0.0 support! 🔥

See updated [changelog](https://epmatt.github.io/awesome-ha-blueprints/docs/blueprints/controllers/ikea_e1743) here.

We will now replicate changes and bring Z2M 2.0.0 support to all controllers.

**Comment by EPMatt** [comment]
Hey everyone! I was finally able to review @yarafie's contribution #677  and merge changes to main. Fixed blueprints are not available on main, so I'm marking this issue as resolved 🚀 If you're still experiencing problems, feel free to open a new issue.

Huge thanks to anyone who contributed to the discussion 🔥
